value/value.go

```diff
@@ -122,11 +120,32 @@ func (i *IntValue) String() string {
 	return fmt.Sprintf("%d", *i)
 }
 
-type DoubleValue float64
+type DoubleValue struct {
+	Double   float64
+	asString string
+	bin64Enc uint64
+}
 
-func NewDoubleValue(v float64) *DoubleValue {
-	i := DoubleValue(v)
-	return &i
+func NewDoubleUsingFloat(v float64) *DoubleValue {
+	return &DoubleValue{
+		Double:   v,
+		bin64Enc: math.Float64bits(v),
+		asString: strconv.FormatFloat(v, 'f', -1, 64),
+	}
+}
+
+func NewDoubleValue(raw string) (*DoubleValue, error) {
+	val, err := strconv.ParseFloat(raw, 64)
+	if err != nil {
+		return nil, api.Errorf(api.Code_INVALID_ARGUMENT, errors.Wrap(err, "unsupported value type ").Error())
+	}
+
+	i := &DoubleValue{
+		Double:   val,
+		asString: raw,
+		bin64Enc: math.Float64bits(val),
+	}
+	return i, nil
 }
 
 func (d *DoubleValue) CompareTo(v Value) (int, error) {
```

Comment:
Do we save this DoubleValue in the search engine?
How ordering will work?

Reply:
it is bin32 encoding so ordering will work but as it is bin32 encoding it is limited and does not work well for big floats.